pkg/controllermanager/controller/controllerregistration/seedfinalizer/add.go

```diff
@@ -29,5 +36,30 @@ func (r *Reconciler) AddToManager(mgr manager.Manager) error {
 		WithOptions(controller.Options{
 			MaxConcurrentReconciles: 5,
 		}).
+		Watches(
+			&gardencorev1beta1.ControllerInstallation{},
+			handler.EnqueueRequestsFromMapFunc(r.MapControllerInstallationToSeed),
+			builder.WithPredicates(r.ControllerUninstallationPredicate()),
+		).
 		Complete(r)
 }
+
+// MapControllerInstallationToSeed returns a reconcile.Request object for the seed specified in the .spec.seedRef.name field.
+func (r *Reconciler) MapControllerInstallationToSeed(_ context.Context, obj client.Object) []reconcile.Request {
```

Comment:
Could you please add unit tests?

Reply:
1. Does `MapControllerInstallationToSeed` need to be a method of `Reconcile`? If yes we can replace `func (r *Reconciler)` with `func (*Reconciler)`. If no lets make it a function
2. Do we need `MapControllerInstallationToSeed` to be exported? The only reason for this could be for testing but it is too small and simple for a test to give any value. Added value of it being unexported is that it will not be skipped by `unused` linter as `unused` skips exported functions. Alternatively we can just inline it

Reply:
@Kostov6 Please check the rest of the code base - it's done like this at most other places, so perhaps we should strive for consistency here.

Reply:
Ok I see what you are referring to. I was just wondering what value do tests like this bring:
https://github.com/gardener/gardener/blob/48169a21cfbfd5a515a7374c729c9de9111e5bd2/pkg/gardenlet/controller/backupbucket/add_test.go#L88-L92
to small functions like:
https://github.com/gardener/gardener/blob/48169a21cfbfd5a515a7374c729c9de9111e5bd2/pkg/gardenlet/controller/backupbucket/add.go#L92-L94
After reading [Purpose of Unit Tests](https://github.com/gardener/gardener/blob/master/docs/development/testing.md#purpose-of-unit-tests) and [Writing Unit Tests](https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-unit-tests) I'm still missing the point. Preventing regression? Im fine with sticking what's been done through the code base. I think documenting and referencing codebase practices are valuable - it specifies the practice and is especially  useful when onboarding new colleagues so let's just add a sentence or two into the docs giving rationale for tests like these. Can you give a proposal? I'm fine with opening the PR for adding it to the docs

Reply:
I think we started this with functions that are more complex than this one which we wanted to unit-test. While progressing further along the refactoring with `controller-runtime` back then, we simply followed this pattern in all controllers. Some of the logic is trivial (like your example) and a unit test is quasi useless, but so what. It also isn't much of an effort to add it (for the sake of consistency), and if the logic ever becomes more complex, the "unit test infrastructure" is already there to get augmented.

Reply:
> Preventing regression?

Mainly yes. It's a small function that's easy and cheap to test - so why not do it?!

> Can you give a proposal?

Not sure what proposal you expect here. My proposal would be: Not writing tests for newly added code must be justified well.

Reply:
Thanks for providing the historical context. Is it ok to add under `Writing Unit Tests` something along the lines: 
- For common cases like `add_test.go` and `reconciler_test.go` please follow the established testing pattern

Do other cases come to your mind to be included?

Reply:
> Not sure what proposal you expect here. My proposal would be: Not writing tests for newly added code must be justified well.

I wrote my comment before reading your reply. I actually like your proposal and see a benefit in adding it to the docs. It's just that I don't think there is a general common agreement on testing among devs so "obvious" for some is "bad" for others